Realm/Realm.Weaver/RealmWeaver.cs

```diff
@@ -256,16 +286,25 @@ private WeaveTypeResult WeaveGeneratedType(TypeDefinition type)
             var interfaceType = _moduleDefinition.GetType($"{@namespace}.Generated", interfaceName);
 
             var persistedProperties = new List<WeavePropertyResult>();
+            var backingFields = new HashSet<MetadataToken>();
 
             // We need to weave all (and only) the properties in the accessor interface
             foreach (var interfaceProperty in interfaceType.Properties)
             {
                 var prop = type.Properties.First(p => p.Name == interfaceProperty.Name);
-
                 try
                 {
+                    // Stash and remove the backing field before weaving as it depends on get method.
+                    var backingField = prop.GetBackingField();
```

Comment:
So if you look at source code of `.GetBackingField` it looks at `prop.GetMethod` which does get cleared when weaving, which is why I am doing this that way, maybe there is a better way to structure it though